lib/geolookup/usa/areacodes.rb

```diff
@@ -3,14 +3,103 @@ module USA
     module AreaCodes
 
       AREA_CODE_META_INFO_FILE = 'AREA_CODE_META_INFO.yml'
+      AREACODE_POPULATION_FILE = 'CITY_STATE_COUNTY_POPULATION.csv'
 
       def self.to_h
         @area_codes_hash ||= Geolookup.load_hash_from_file(AREA_CODE_META_INFO_FILE)
       end
 
+      def self.details
+        @records ||= Geolookup.load_list_from_csv(AREACODE_POPULATION_FILE).map do |r|
+          Detail.new(areacode: r[0].to_i,
+                     city: r[1],
+                     statecode: r[2].to_i,
+                     countycode: r[3].to_i,
+                     population: r[4].to_i)
+        end
+      end
+
       def self.find(area_code)
         to_h[area_code.to_s] || {}
       end
+
+      # Return a list of detail records for areacode,
+      # sorted by population in descending order.
```

Comment:
would there ever be a need to return the population number ever?

Reply:
The population number is part of the _Detail_ record, it can be accessed from the response.